contracts/asset-proxy/contracts/src/bridges/DydxBridge.sol

```diff
@@ -0,0 +1,186 @@
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "../interfaces/IERC20Bridge.sol";
+import "../interfaces/IDydxBridge.sol";
+import "../interfaces/IDydx.sol";
+
+
+contract DydxBridge is
+    IERC20Bridge,
+    IDydxBridge,
+    DeploymentConstants
+{
+
+    /// @dev Callback for `IERC20Bridge`. Deposits or withdraws tokens from a dydx account.
+    ///      Notes:
+    ///         1. This bridge must be set as an operator of the input dydx account.
+    ///         2. This function may only be called in the context of the 0x Exchange.
+    ///         3. The maker of the 0x order must be the dydx account owner.
+    ///         4. To deposit into dydx use this function in the `takerAssetData`.
+    ///            (The `to` address is the 0x order maker and dydx account owner).
+    ///         5. To withdraw from dydx use this function in the `makerAssetData`.
+    ///            (The `from` address is the 0x order maker and dydx account owner).
+    /// @param from The sender of the tokens.
+    /// @param to The recipient of the tokens.
+    /// @param amount Minimum amount of `toTokenAddress` tokens to deposit or withdraw.
+    /// @param encodedBridgeData An abi-encoded `BridgeData` struct.
+    /// @return success The magic bytes if successful.
+    function bridgeTransferFrom(
+        address,
+        address from,
+        address to,
+        uint256 amount,
+        bytes calldata encodedBridgeData
+    )
+        external
+        returns (bytes4 success)
+    {
+        // Ensure that only the `ERC20BridgeProxy` can call this function.
+        require(
+            msg.sender == _getERC20BridgeProxyAddress(),
+            "DydxBridge/ONLY_CALLABLE_BY_ERC20_BRIDGE_PROXY"
+        );
+
+        // Decode bridge data.
+        (BridgeData memory bridgeData) = abi.decode(encodedBridgeData, (BridgeData));
+
+        // Cache dydx contract.
+        IDydx dydx = IDydx(_getDydxAddress());
+
+        // Create dydx action.
+        IDydx.AccountInfo[] memory accounts = new IDydx.AccountInfo[](1);
+        IDydx.ActionArgs[] memory actions = new IDydx.ActionArgs[](1);
+        if (bridgeData.action == BridgeAction.Deposit) {
+            // We interpret `to` as the 0x order maker and `from` as the taker.
+            accounts[0] = IDydx.AccountInfo({
+                owner: to,
```

Comment:
Q: What happens if the someone creates an order with a deposit dydx asset data in the maker or taker *fee* asset data? Can filling the order effectively force the fee recipient into performing a deposit to their account?

Reply:
Ah yeah, that would be annoying. It's probably worth using the more complex logic to avoid people doing this. I can revert to the previous version that only allowed operations on the account owned by the `from` address.